README.md

```diff
@@ -33,16 +33,46 @@ fmt.Println(lang)
 lang := enry.GetLanguage("foo.cpp", "<cpp-code>")
 ```
 
-Developmemt
+Development
 -----------
-*enry* re-uses parts of original [linguist](https://github.com/github/linguist) especially data in `languages.yml` to generate internal data structures. In oreder to update to latest upstream run
+*enry* re-uses parts of original [linguist](https://github.com/github/linguist) especially data in `languages.yml` to generate internal data structures. In order to update to latest upstream run
 
     make clean code-generate
 
 To run the tests
 
     make test
 
+Benchmark
+---------
+Enry's language detection has been compared  with Linguist's language detection. In order to do that, linguist's project directory [*linguist/samples*](https://github.com/github/linguist/tree/master/samples) was used as a set of files to run benchmarks against.
+
+ Following results were obtained:
+
+![histogram](https://raw.githubusercontent.com/src-c/enry/master/benchmark/histogram/distribution.jpg)
+
+The histogram represents the number of files for which spent time in language detection was in the range of the time interval indicated in x axis.
+
+So reviewing the comparison enry/linguist, you can see the most of the files were detected in less time than linguist does.
+
+We detected some few cases enry turns slower than linguist due to Golang regexp while Ruby uses [oniguruma](https://github.com/kkos/oniguruma) as a regular expression engine (a regular expression library written in C).
+
+You can find scripts and additional information (as software and hardware used, and benchmark's results per sample file) in [*benchmark*](https://github.com/src-d/enry/tree/master/benchmark) directory.
+
+If you want to reproduce the same experiment you can run:
+
+    benchmark/run.sh
+
+from the root's project directory and It runs benchmarks for enry and linguist, parse the output, create csv files and create a histogram (you must have installed [gnuplot](http://gnuplot.info) in your system to get the histogram). It can take to much time, so to run local benchmarks to take a quick look you can run either:
+
+    make benchmarks
```

Comment:
`benchmark` is used in singular in some places and in plural in others. It should be consistent (plural makes more sense I think).